setup.sh

```diff
@@ -1,45 +1,109 @@
 #!/bin/bash
 
+#  Copyright 2023 DATA @ UHN. See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+# Check the user's authorship configuration
+if ! $(git config user.name 1> /dev/null) || ! $(git config user.email 1> /dev/null)
+then
+  whiptail --backtitle "New CARDS repository setup" --msgbox "You must configure the user name and email address with \"git config user.name Some Name\" and \"git config user.email somebody@email.com\" and then rerun this script. Exiting." 9 78
+  exit 1
+fi
+
+if ! whiptail --backtitle "New CARDS repository setup" --title "Please confirm git authorship details" --yesno "Committing as $(git config user.name) <$(git config user.email)>, OK?" 8 78
+then
+  whiptail --backtitle "New CARDS repository setup" --msgbox "You can configure the user name and email address with \"git config user.name Some Name\" and \"git config user.email somebody@email.com\" and then rerun this script. Exiting." 9 78
+  exit 1
+fi
+
 # Ask about the generic CARDS version to use
 
-# Get the latest CARDS version from Nexus
-CARDS_VERSION=$(curl -s https://nexus.phenotips.org/nexus/content/repositories/releases/io/uhndata/cards/cards-modules/maven-metadata.xml | grep --max-count=1 '<release>' | cut '-d>' -f2 | cut '-d<' -f1)
-if [[ -z $CARDS_VERSION ]]
+# Get the available CARDS versions from Maven Central
+declare -a RELEASED_VERSIONS=( $(curl -s https://repo1.maven.org/maven2/io/uhndata/cards/cards-modules/maven-metadata.xml | grep '<version>' | tac | cut '-d>' -f2 | cut '-d<' -f1) )
+declare -i RELEASED_VERSIONS_COUNT=${#RELEASED_VERSIONS[*]}
+# Look on the local disk
+LOCAL_CARDS_VERSION=$(docker run --rm  --entrypoint /bin/sh cards/cards:latest -c "ls /root/.m2/repository/io/uhndata/cards/cards-parent/ | grep SNAPSHOT")
```

Comment:
I think it would be best to first check that `cards/cards:latest` is available locally and only run this command if such is the case. Otherwise, this command will attempt to pull `cards/cards:latest` from Docker Hub if no `cards/cards:latest` image exists locally. Please see https://github.com/data-team-uhn/cards/blob/dev/Utilities/Packaging/Docker/build_self_contained.sh#L28-L34 for an example of how we do this check.

Reply:
Done.